app/src/main/java/io/gnosis/safe/ui/base/BaseStateViewModel.kt

```diff
@@ -15,9 +20,11 @@ data class AppDispatchers(
 )
 
 abstract class BaseStateViewModel<T>(private val dispatchers: AppDispatchers) : ViewModel() where T : BaseStateViewModel.State {
-    abstract val state: LiveData<T>
 
-//    private val errorHandler = SimpleLocalizedException.networkErrorHandlerBuilder(context).build()
+    val state: LiveData<T> = liveData {
+        for (event in stateChannel.openSubscription())
```

Comment:
We could still provide a callback here. As in the legacy app in some places behaviour is injected here. Like an `open fun onStateSubscribed()` with an empty default implementation to not have to do it everywhere.